Update datatable

eric-burel opened the pull request:
Brings extremely interesting improvement to the datatable:
- a checkbox to select items
- a CSV export
- a deletion button

@EloyID I've merged your datatable enhancement here. Do you have enough time to create a demo in the Starter "example-form" package? I've tested the code a bit in Storybook but I can check every features you added. You could enable them in the example to show a full featured datatable.

Apollinaire committed:
Add useSignout hook to the accounts package

eric-burel committed:
draft auth query

eric-burel committed:
first experiment login an user through graphql, can get token

eric-burel committed:
add a logout handler

eric-burel committed:
move queries to mutations for auth

eric-burel committed:
fix login

eric-burel committed:
first try with cors

eric-burel committed:
enable CORS whitelisting

eric-burel committed:
add link to simple rest for auth

EloyID committed:
add Material UI DatatableFilterCheckboxes

EloyID committed:
add showDelete option to Datatable

EloyID committed:
pass fragment to useDelete

EloyID committed:
DeleteButton creation

EloyID committed:
add selection system to datatable

EloyID committed:
bugfix in datatable selection system

EloyID committed:
merge es_ES

EloyID committed:
change selectable to showSelect in Datatable

EloyID committed:
linking datatable filters with csv export

EloyID committed:
link csvexport with selection in datatable

EloyID committed:
DeleteButton support many documents deletion

EloyID committed:
add many delete button in datatable

EloyID committed:
can add custom components above the Datatable

EloyID committed:
split datatable in different files

EloyID committed:
split datatable in different files 2

EloyID committed:
split datatable in different files 2

EloyID committed:
split datatable in different files 3

EloyID committed:
split datatable in different files 4 saving with prettier

EloyID committed:
adapt material ui datatable to the bootstrap features

EloyID committed:
adapt material ui datatable to the bootstrap features 2

EloyID committed:
adapt material ui datatable to the bootstrap features 3

EloyID committed:
material ui datatable comments cleaning

EloyID committed:
include delete and export functions in datatable material ui

EloyID committed:
add csv export button for material ui

EloyID committed:
bugfix csvexportbutton

EloyID committed:
create material ui deletebutton

EloyID committed:
set showdelete false default in datatable

EloyID committed:
switch <a> to button so no need of javascript:void(0) which throws warning

EloyID committed:
move muicheckboxgroup proptypes name and options to inputProperties and name is not required anymore

EloyID committed:
bugfix datatable material ui, if filtering gives no result [] we still show the content or the filters will dissapear

EloyID committed:
bugfix material ui datatable not getting value from url

EloyID committed:
bugfix for the input api so it can accept more than one input for a samme field

eric-burel committed:
Merge branch 'enhancement/datatable' of https://github.com/live-for-good/Vulcan into live-for-good-enhancement/datatable

eric-burel committed:
add react-csv dependency

eric-burel committed:
do not fail on empty selection

eric-burel committed:
fix datatable item selection

eric-burel committed:
merge mui

eric-burel committed:
add documents to CardItemSwitcher

eric-burel committed:
Merge branch 'live-for-good-enhancement/datatable' into feature/datatable-update

eric-burel committed:
Merge pull request #2554 from live-for-good/bugfix/checkboxmuigroup

move muicheckboxgroup proptypes name and options to inputProperties a…

EloyID commented:
I will try to create a demo (lot of work), but the filters won't work correctly without https://github.com/VulcanJS/Vulcan/pull/2552

eric-burel commented:
Ok I'll try to merge it within a few days

eric-burel committed:
Merge branch 'bugfix/input_api' of https://github.com/live-for-good/Vulcan into live-for-good-bugfix/input_api

eric-burel committed:
add test for mongoParams

eric-burel committed:
Merge branch 'live-for-good-bugfix/input_api' into devel

eric-burel committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

eric-burel commented:
@EloyID  #2552 is now merged into devel. I've also pushed the commit on "feature/datatable" (I had forgotten to push sorry)

SachaG committed:
Internationalize field option labels

SachaG committed:
Update fourseven:scss package versions

SachaG committed:
Work on Likert component

SachaG committed:
comment -> document

SachaG committed:
Share getGraphQLType code with client

SachaG committed:
Move field-specific data loading to new FormComponentLoader component

SachaG committed:
Do not handle options at form level anymore since we don't have field-specific data yet

SachaG committed:
Add support for _like; allow _in to be empty

SachaG committed:
Add AutocompleteMultiple input

SachaG committed:
Finish autocomplete

SachaG committed:
Autocomplete clean-up; move templates with other GraphQL query templates

EloyID committed:
Enable OpenCRUD backwards compatibility

When using onEdit we should get as args : cf. https://github.com/VulcanJS/Vulcan/blob/51acfa7ffcec844cab882f661ca314df19f56fa4/packages/vulcan-lib/lib/server/mutators.js#L193 
- modifier
- oldDocument
- currentUser
- document
Nevertheless, in the current version we get document twice which prevents us from verifying oldDocument properties. I think at some commit document became oldDocument and newDocument became document.

SachaG committed:
Move decorators to core package

SachaG committed:
Add error tooltips to mutation buttons

SachaG committed:
siteData query resolver should not be capitalized

SachaG committed:
Array fields should also support _in (remove contain for now)

SachaG committed:
Do not handle option queries without "query" keyword anymore;

SachaG committed:
Refactoring newsletter package; update Mailchimp API integration to v3.0

wiiiimm committed:
Use Meteor's version of npm over system default

close #2506

eric-burel committed:
Merge pull request #2560 from williamli/patch-1

Use Meteor's version of npm over system default

eric-burel committed:
fix for same origin

eric-burel committed:
add additional mutations for email verification and password creation

eric-burel committed:
added signup mutation

eric-burel committed:
fix signup typo

kevinashworth committed:
Don't run `options` function until `loading` is false, and `data` is populated
This structure is also in keeping with the code in Apollo example. See https://www.apollographql.com/docs/react/api/react-hooks/#example

SachaG committed:
Make autocomplete guess resolver from relation; backwards compatibility support for resolveAs.type vs resolveAs.typeName; specify Apollo 3.1.0

SachaG committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

eric-burel committed:
tested and fixed mutations

manual test

eric-burel committed:
Merge branch 'devel' into feature/graphql-account

eric-burel committed:
fix tests

eric-burel committed:
Merge pull request #2530 from VulcanJS/feature/graphql-account

Account/auth features available outside DDP using GraphQL

eric-burel committed:
Wrap the loader with the same div as the component

eric-burel committed:
Merge pull request #2561 from kevinashworth/bugfix/loading-options

Don't run `options` function until `loading` is false, and `data` is populated

eric-burel committed:
Merge pull request #2559 from EloyID/patch-2

Enable OpenCRUD backwards compatibility

eric-burel committed:
Update and rename useSignout.js to useMeteorLogout.js

eric-burel committed:
Update main_client.js

eric-burel committed:
Update main_server.js

eric-burel committed:
Merge pull request #2521 from Apollinaire/useSignout

Add useSignout hook to the accounts package

eric-burel committed:
add apollo options to sample settings

eric-burel committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

kevinashworth committed:
Update react-helmet to 6.0.0

kevinashworth committed:
Use Helmet as a named import

`react-helmet` 6 no longer uses a named export - see https://github.com/nfl/react-helmet/wiki/Upgrade-from-5.x.x----6.x.x-beta

eric-burel committed:
can list fields that need the document to be checked

eric-burel committed:
exclude ambiguous fields from first check

eric-burel committed:
correct filter check for custom permissions

eric-burel committed:
tests passing for both resolvers

eric-burel committed:
fixed => filter out documents instead of failing

eric-burel committed:
cleanup tests a bit

eric-burel committed:
Merge pull request #2563 from kevinashworth/feature/update-react-helmet

feature/update react-helmet

SachaG committed:
Add submitCallback prop to MutationButton

SachaG committed:
Clean up vulcan:newsletter package

SachaG committed:
Add locale to lang attribute on HTML element

SachaG committed:
Pass GraphQL errors to useMulti2 result

SachaG committed:
Throw error when autocompletePropertyName is missing

SachaG committed:
New PaginatedList component

SachaG committed:
Token -> Cell

SachaG committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

eric-burel committed:
fix install with Meteor 1.10

SachaG committed:
Backoffice readme

SachaG committed:
Use useLazyQuery for Autocomplete components

SachaG committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

SachaG committed:
Improve clear form field button

SachaG committed:
Clean up form props passing a bit; make mergeWithComponent work more like other components by accepting `components` prop

SachaG committed:
Refactor FormComponent so that character counts update on any value change; queryLoading -> loading; Add new FormInputLoading wrapper component

eric-burel committed:
update node-sass to fix failing build

eric-burel committed:
Merge pull request #2564 from VulcanJS/bugfix/2550-readable-fields-with-document

Bugfix/2550 readable fields with document

SachaG committed:
Pass input options to Connectors.get

SachaG committed:
Do not allow empty input for single resolver

SachaG committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

# Conflicts:
#	packages/vulcan-lib/lib/server/default_resolvers.js

SachaG committed:
Handle multiple values being updated better (don't use prevState because it doesn't mutate after each loop iteration)

SachaG committed:
Do not use context to get document, pass it as prop instead

SachaG committed:
v1.15.0

eric-burel committed:
bump version

eric-burel committed:
update versions

SachaG committed:
Merge branch 'release/1.15.0' into devel

# Conflicts:
#	packages/vulcan-backoffice/package.js
#	packages/vulcan-core/package.js
#	packages/vulcan-test/package.js
#	packages/vulcan-ui-material/package.js
#	packages/vulcan-voting/package.js

eric-burel committed:
update versions

SachaG committed:
Fix _contains

SachaG committed:
currentValues -> document; pass down submitForm; refactor getProperties and add onKeyPress event handler to make enter key work for submitting forms

SachaG committed:
Merge branch 'devel' of https://github.com/VulcanJS/Vulcan into devel

kevinashworth committed:
Reverse commit 594bf423bc86aa83061b53866f6fcefa6415be03

SmartForm changes from a week ago cause an error — cannot create or update nested inputs, like those in the Addresses subschema of VulcanStarter example-forms.

Commit 594bf423bc86aa83061b53866f6fcefa6415be03 introduced the error. When I reverse just this one commit, I am once again able to update nested fields.

SachaG committed:
Merge pull request #2574 from kevinashworth/bugfix/smartform-nested-inputs

bugfix - SmartForm cannot update nested input fields

Commit:
change Mui to expect handleChange in props

SachaG committed:
Merge pull request #2577 from yairtal/bugfix/MuiSelect-props-onChange

change Mui to expect handleChange in props

eric-burel committed:
bump version

eric-burel committed:
Merge branch 'feature/datatable-update' into devel

eric-burel committed:
commented CSV export for now
